src/main/kotlin/com/coop/technologies/kotlinInsightApi/InsightAPI.kt

```diff
@@ -159,80 +160,93 @@ object InsightCloudApi {
         val kobj = Class.forName(clazz.name).kotlin
         val result = kobj.primaryConstructor
             ?.parameters
-            ?.fold(emptyMap<KParameter, Any?>()) { map, parameter ->
+            ?.map { parameter ->
                 var value = values[parameter.name?.capitalize()]
                 val reference = references[parameter.name?.capitalize()]
                 val definedClass = fields[parameter.name?.capitalize()]
-                var result = value
-                when (definedClass) {
-                    Int::class.java -> result = value.toString().toInt()
-                    Float::class.java -> result = value.toString().toFloat()
-                    Double::class.java -> result = value.toString().toDouble()
-                    Boolean::class.java -> result = value.toString().toBoolean()
-                }
-                if (reference == null && definedClass == List::class.java) {
-                    val outClass =
-                        Class.forName(parameter.type.arguments.first().type!!.javaType.typeName!!)
-                    if (value == null) {
-                        value = emptyList<String>()
-                    }
-                    when (outClass) {
-                        Integer::class.java -> result = (value as List<String>).map { it.toInt() }
-                        Float::class.java -> result = (value as List<String>).map { it.toFloat() }
-                        Double::class.java -> result = (value as List<String>).map { it.toDouble() }
-                        Boolean::class.java -> result =
-                            (value as List<String>).map { it.toBoolean() }
+                val result = when {
+                    definedClass == Int::class.java -> value.toString().toInt()
+                    definedClass == Float::class.java -> value.toString().toFloat()
+                    definedClass == Double::class.java -> value.toString().toDouble()
+                    definedClass == Boolean::class.java -> value.toString().toBoolean()
+                    definedClass == String::class.java -> {
+                        try {
+                            value as String?
```

Comment:
Is this try catch only relevant for String and not for the other Classes? E.g. if you have an optional Int in the data model?

Reply:
This behavior I only found with strings being empty, float values at least didn't show similar issues.
The reason for the try catch is that, for some odd reason Insight will return an empty string as emptyArray, which in turn can't be cast to string.